nltools/data/brain_data.py

```diff
@@ -1839,52 +1800,58 @@ def align(self, target, method='procrustes', n_features=None, axis=0,
 
         '''
 
-        source = self.copy()
-        common = target.copy()
-
-        target = check_brain_data(target)
-
         if method not in ['probabilistic_srm', 'deterministic_srm', 'procrustes']:
             raise ValueError("Method must be ['probabilistic_srm','deterministic_srm','procrustes']")
+        
+        source = self.copy()
+        data1 = self.data.copy()
 
-        data1 = source.data.T
-        data2 = target.data.T
+        if method == 'procrustes':
+            target = check_brain_data(target)
+            data2 = target.data.copy()
+            
+            # pad columns if different shapes
+            sizes_1 = [x.shape[1] for x in [data1, data2]]
+            C = max(sizes_1)
+            y = data1[:, 0:C]
+            missing = C - y.shape[1]
+            add = np.zeros((y.shape[0], missing))
+            data1 = np.append(y, add, axis=1)
+        else:
+            data2 = target.copy()
 
         if axis == 1:
             data1 = data1.T
             data2 = data2.T
 
-        out = dict()
+        out = {}
         if method in ['deterministic_srm', 'probabilistic_srm']:
-            if n_features is None:
-                n_features = data1.shape[0]
-            if method == 'deterministic_srm':
-                srm = DetSRM(features=n_features, *args, **kwargs)
-            elif method == 'probabilistic_srm':
-                srm = SRM(features=n_features, *args, **kwargs)
-            srm.fit([data1, data2])
-            source.data = srm.transform([data1, data2])[0].T
-            common.data = srm.s_.T
-            out['transformed'] = source
-            out['common_model'] = common
-            out['transformation_matrix'] = srm.w_[0]
+            if data2.shape[0] != data1.shape[0]:
+                raise ValueError("The number of timepoints(TRs) does not match the model.")
+
+            A = data1.T.dot(data2)
+
+            # # Solve the Procrustes problem
```

Comment:
Wait something is wrong here. You're not actually using `SRM` or `DetSRM` at all despite them being imported at the top of the file?? Did those get accidentally removed in a commit? I even see them in the current version of nltools prior to this PR.

Reply:
I thought I deleted the import statements. 

I now just rotate the single subject to the common_model. Pretty much exactly what SRM does internally. much simpler and faster than what I was doing before, which was estimating a new SRM model.

Reply:
just checked import statements are deleted.

Reply:
Wait I"m looking at the SRM source code and this only partially correct for the _deterministic_ SRM. You're missing an initialization step here: https://github.com/brainiak/brainiak/blob/master/brainiak/funcalign/srm.py#L818

Also the _probabilistic_ SRM uses and EM optimization method here: https://github.com/brainiak/brainiak/blob/master/brainiak/funcalign/srm.py#L465

So it's still not quite accurate to use the _same_ operation for both version in our code no?

Reply:
It looks like your version is closest to the low RAM/fast SRM _only_. I feel like we should be clear about that or implement the other methods properly: https://github.com/brainiak/brainiak/blob/78370dff01829a79bb0498a50594187b57ca836b/brainiak/funcalign/fastsrm.py#L1006